onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -238,6 +241,36 @@ class PlannerImpl {
     return false;
   }
 
+  bool FindReusableGraphInput(const onnxruntime::Node& node, int output_arg_num, MLValueIndex* reusable_input) {
+    // skip if this is main graph
+    if (parent_node_) {
+      // only check a single layer. ie. an Identity/Dropout node connecting graph's input and output
+      auto& graph_inputs = graph_viewer_.GetInputs();
+      auto& graph_outputs = graph_viewer_.GetOutputs();
+
+      if (FindReusableInput(node, output_arg_num, reusable_input)) {
+        auto& arg = ml_value_info_.at(Buffer(*reusable_input)).p_def_site;
+        auto it = std::find(graph_inputs.cbegin(), graph_inputs.cend(), arg);
+        if (it != graph_inputs.end()) {
+          auto graph_input_index = std::distance(graph_inputs.cbegin(), it);
+          ORT_ENFORCE(graph_input_index >= 0 && static_cast<size_t>(graph_input_index) < graph_inputs.size());
+          auto graph_output_index = std::distance(graph_outputs.cbegin(),
+                                                  std::find(graph_outputs.cbegin(), graph_outputs.cend(), node.OutputDefs()[output_arg_num]));
```

Comment:
>node.OutputDefs()[output_arg_num]) [](start = 106, length = 34)

This should probably be outside the iteration so OutputDefs() and operator[] aren't called every time. Will probably get optimized out, but no need to rely on that.